pkg/network/tracer/connection/kprobe/tracer.go

```diff
@@ -218,6 +218,22 @@ func loadTracerFromAsset(buf bytecode.AssetReader, runtimeTracer, coreTracer boo
 		}
 	}
 
+	// Replace with map error telemetry and helper error telemetry maps with
+	// per-cpu maps if kernel supports them.
+	if features.HaveMapType(ebpf.PerCPUHash) == nil {
```

Comment:
What happens when per-cpu is not supported? Won't the init/lookup functions fail?

Reply:
It would. I have to add support for that case.

Reply:
Since we already require 4.14 as the minimum version for ebpf telemetry, I also added per-cpu maps to be required as well, since they are already present since 4.7.

This way we do not have to handle the case when per-cpu is not supported.

Reply:
The telemetry code is still loaded in environments that do not support ebpf telemetry. The ebpf code itself has to pass the verifier in all cases. We do load time adjustments when we actually want the telemetry to run.
The adjustments are:
1. Patching the xadd instructions, so the verifier passes.
2. Changing error telemetry map types from hash maps to per-cpu maps.